src/network/networkd-mptcp.c

```diff
@@ -0,0 +1,337 @@
+                if (r < 0)
+                        return log_link_error_errno(link, r, "Could not append MPTCP_PM_ADDR_ATTR_ID attribute: %m");
+        }
+
+        r = sd_netlink_message_append_s32(req, MPTCP_PM_ADDR_ATTR_IF_IDX, link->ifindex);
+        if (r < 0)
+                return log_link_error_errno(link, r, "Could not append MPTCP_PM_ADDR_ATTR_IF_IDX attribute: %m");
+
+        if (in_addr_is_null(mp_tcp->family, &mp_tcp->address) == 0) {
+                r = sd_netlink_message_append_u16(req, MPTCP_PM_ADDR_ATTR_FAMILY, mp_tcp->family);
+                if (r < 0)
+                        return log_link_error_errno(link, r, "Could not append MPTCP_PM_ADDR_ATTR_FAMILY attribute: %m");
+
+                if (mp_tcp->family == AF_INET)
+                        r = sd_netlink_message_append_in_addr(req, MPTCP_PM_ADDR_ATTR_ADDR4, &mp_tcp->address.in);
+                else
+                        r = sd_netlink_message_append_in6_addr(req, MPTCP_PM_ADDR_ATTR_ADDR6, &mp_tcp->address.in6);
+                if (r < 0)
+                        return log_link_error_errno(link, r, "Could not append MPTCP_PM_ADDR_ATTR_ADDR4 / MPTCP_PM_ADDR_ATTR_ADDR6 attribute: %m");
+        }
+
+        r = sd_netlink_message_close_container(req);
+        if (r < 0)
+                return log_link_error_errno(link, r, "Could not close MPTCP_PM_ATTR_ADDR container: %m");
+
+        r = netlink_call_async(link->manager->genl, NULL, req, mp_tcp_handler,
+                               link_netlink_destroy_callback, link);
+        if (r < 0)
+                return log_link_error_errno(link, r, "Could not send rtnetlink message: %m");
+
+        link_ref(link);
+        link->mp_tcp_messages++;
+
+        return 0;
+}
+
+static int mp_tcp_limit_handler(sd_netlink *rtnl, sd_netlink_message *n, Manager *m) {
+        int r;
+
+        assert(n);
+        assert(m);
+
+        r = sd_netlink_message_get_errno(n);
+        if (r < 0 && r != -EINVAL) {
+                log_error_errno(r, "Could not set up MPTCP limits: %m");
+                return 1;
+        }
+
+        log_debug("MPTCP limit configured");
+
+        return 1;
+}
+
+int mp_tcp_configure_limit(Manager *m) {
+        _cleanup_(sd_netlink_message_unrefp) sd_netlink_message *req = NULL;
+        int r;
+
+        assert(m);
+        assert(m->rtnl);
+
+        if (m->mp_tcp_subflows <= 0 && m->mp_tcp_add_addr_accepted <= 0)
+                return 0;
+
+        r = sd_genl_message_new(m->genl, SD_GENL_MPTCP, MPTCP_PM_CMD_SET_LIMITS, &req);
+        if (r < 0)
+                return log_error_errno(r, "Failed to allocate generic netlink message: %m");
+
+        r = sd_netlink_message_append_u32(req, MPTCP_PM_ATTR_RCV_ADD_ADDRS, m->mp_tcp_subflows);
+        if (r < 0)
+                return log_error_errno(r, "Could not append MPTCP_PM_ATTR_RCV_ADD_ADDRS attribute: %m");
+
+        r = sd_netlink_message_append_u32(req, MPTCP_PM_ATTR_SUBFLOWS, m->mp_tcp_add_addr_accepted);
```

Comment:
Attributes and variables do not match...

Reply:
Can you explain what you mean here